modules/generator/registry/native_histogram.go

```diff
@@ -132,6 +150,31 @@ func (h *nativeHistogram) newSeries(labelValueCombo *LabelValueCombo, value floa
 
 	h.updateSeries(newSeries, value, traceID, multiplier)
 
+	lbls := labelValueCombo.getLabelPair()
+	lb := labels.NewBuilder(make(labels.Labels, 1+len(lbls.names)))
+
+	// set series labels
+	for i, name := range lbls.names {
+		lb.Set(name, lbls.values[i])
+	}
+	// set external labels
+	for name, value := range h.externalLabels {
+		lb.Set(name, value)
+	}
+
+	lb.Set(labels.MetricName, h.metricName)
+
+	newSeries.labels = lb.Labels()
+	newSeries.lb = lb
```

Comment:
ok, for clarity this becomes the label builder for all `le` series? it seems like that's basically the only place it's used. 

just concerned that labels from one series will "leak" into another if we share this label builder too much. it seems like the same label builder is on the main histogram and passed to all the series?

Reply:
Yes that's correct.  Since Labels() doesn't return a pointer, then we just get whatever it had at that time, which I didn't realize until I started poking on this PR.  Here is why I think its okay.
For [count](https://github.com/zalegrala/tempo/blob/1cb964399993c7f8283a1b2e830030215c5aee96/modules/generator/registry/native_histogram.go#L169) and [sum](https://github.com/zalegrala/tempo/blob/1cb964399993c7f8283a1b2e830030215c5aee96/modules/generator/registry/native_histogram.go#L173) metrics, we capture the labels at init.  Then the only place that we use the stored label builder is in the `classicHistograms`, which you noted.  Then we set the name before we use any of the labels [here](https://github.com/zalegrala/tempo/blob/1cb964399993c7f8283a1b2e830030215c5aee96/modules/generator/registry/native_histogram.go#L336), then loop over all the buckets and set their `le` label [here](https://github.com/zalegrala/tempo/blob/1cb964399993c7f8283a1b2e830030215c5aee96/modules/generator/registry/native_histogram.go#L344), same for the +Inf bucket.  Then we clean up at the bottom [here](https://github.com/zalegrala/tempo/blob/1cb964399993c7f8283a1b2e830030215c5aee96/modules/generator/registry/native_histogram.go#L381).

One improvement I can see is that perhaps we want to defer that delete, since there are opportunities to return early which wouldn't delete the `le` label, but this probably doesn't matter because we set it before we ever use it, so even the delete at the bottom is probably unncessary.  (I say this before running the tests).

Reply:
Nice digging. Yup, the .Labels() method copies the internal slice instead of returning it so this is safe.